syncs/syncs.go

```diff
@@ -160,20 +160,29 @@ func (s Semaphore) Release() {
 // at a relatively high frequency.
 // This must not be shallow copied.
 type Map[K comparable, V any] struct {
+	// alreadyLocked reports whether a write-lock is already held.
+	// If so, we avoid acquiring any locks on mu.
+	// This is used by RangeMutable to allow mutations during iteration.
+	alreadyLocked bool
```

Comment:
I can't convince myself this is safe. cc @bradfitz

Reply:
This field is only ever set in a new `Map`, so there should be raciness with regard to this field itself.

With regard to the data protected by `Map.mu`, the `RangeMutable` method